### PR TITLE
[secrets sync] add more thorough Secrets Sync landing page tests

### PR DESCRIPTION
**Pull request by noelledaley**
### :hammer_and_wrench: Description
Adds tests & reorganizes existing tests to ensure we are covering the Secrets Sync / Overview page when:
- [x] OSS
- [x]  Enterprise
    - [x] has feature in license
    - [x] does not have feature in license
    - [x] feature is activated
        - [x] there are destinations
        - [x] there are no destinations
        - [x] activation workflow works
        - [x] activation workflow handles errors
    - [x] feature is not activated

### 🔗 Links
Refer to the Whimsical diagram for another example of all possible states that the page could have. I used this as a reference when outlining all of the tests.

### :camera_flash: Screenshots
Note the tests now being organized in modules that make it clearer which scenario we're testing
updatead tests

### :building_construction: How to Build and Test the Change
Take a look at the following test files and make sure they cover the scenarios outlined above:
- `tests/acceptance/sync/secrets/overview-test.js`
- `tests/integration/components/sync/secrets/page/overview-test.js`

### CODE REVIEW

File: ui/tests/helpers/sync/sync-selectors.js
Comment (noelledaley) [inline ui/tests/helpers/sync/sync-selectors.js:56]: this wasn't used anywhere

File: ui/tests/acceptance/sync/secrets/overview-test.js
Comment (noelledaley) [inline ui/tests/acceptance/sync/secrets/overview-test.js:46]: same test as before, just making it clearer that we're testing the whole secrets sync activation workflow. the [integration test below](https://github.com/hashicorp/vault/pull/26196/files#diff-0d41baf9a55f19f57c3a14b68fea9d5b9247fc0c77e4e1dc9e5da7b5ea34bea1R111) covers all of the dom assertions about making sure the banner and modal content are accurate.

overall, the changes i made across these two files were to keep them focused on:
- acceptance test: covering user interactions / end to end workflows / navigating across routes
- integrations test: "does this component look right given XYZ state"

File: ui/tests/acceptance/sync/secrets/overview-test.js
Comment (noelledaley) [inline ui/tests/acceptance/sync/secrets/overview-test.js:99]: didn't think we needed these extra dom assertions since the integration test covers all of those things being present. this way the two tests in this module are more specifically focused on making sure the requests are made to the right namespace (and not worried about the banner/modal)

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (noelledaley) [inline ui/tests/integration/components/sync/secrets/page/overview-test.js:68]: i apologize for how hard these changes are to read through. much of it is me reorganizing existing tests so it's clearer which scenarios we're covering. here's a helpful screenshot of how the test modules are organized. open to feedback!

Screenshot 2024-03-27 at 7 53 19 PM

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (noelledaley) [inline ui/tests/integration/components/sync/secrets/page/overview-test.js:212]: the complexity of this module is a signal that this chunk of the component should be split into its own component. not gonna do that here, just thinking out loud 🙃

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (noelledaley) [inline ui/tests/integration/components/sync/secrets/page/overview-test.js:270]: this test previously had a title that suggested an empty state would show when there were no destinations, but the content of the test made it seem like it was more about showing an empty state with a “check your policy” message, so i clarified that in the title of the test. 

that being said, i looked at the code and it looks like we show that “check your policy” message even when there isn’t a permissions error — we’d show it if ‘this.destinationMetrics’ is falsey. that seems misleading to me — maybe we should show a different empty state message instead?

File: ui/tests/acceptance/sync/secrets/overview-test.js
Comment (hashishaw) [inline ui/tests/acceptance/sync/secrets/overview-test.js:69]: Maybe we can add a bit more to ensure the flow continues to work? Something like click "Create destination" then you're on the destination create page.

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (hashishaw) [reply ui/tests/integration/components/sync/secrets/page/overview-test.js:68]: I agree with all of this, *except* I like keeping the assertion messages because in the larger tests it is easier to figure out what was happening in the lead-up to the part where it failed. Otherwise the test suite just says "ok" unless it fails, which is not as helpful. But, this is sort of mitigated by the smaller test files, so I think both ways work

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (hashishaw) [inline ui/tests/integration/components/sync/secrets/page/overview-test.js:102]: We should stub features here, right? 
```
this.version.features = ['Secrets Sync']
```

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (hashishaw) [inline ui/tests/integration/components/sync/secrets/page/overview-test.js:86]: Labeling the module this way will make it only run on the enterprise test suite, but since this is an integration test and we're already mocking the relevant pieces elsewhere I would suggest maybe labeling this `ent` so it gets run on OSS and we just mock out the pieces the test is reliant on. That way we get full coverage on every PR rather than waiting until the CE -> Ent merge

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (hashishaw) [reply ui/tests/integration/components/sync/secrets/page/overview-test.js:270]: That seems like a good improvement to make! One thing to keep in mind is that the destinations call will fail with a specific message if the feature is not yet activated (which we can probably skip doing anything because the banner should already exist) and then other types of errors can show a catch-all error banner

File: ui/tests/integration/components/sync/sync-header-test.js
Comment (hashishaw) [inline ui/tests/integration/components/sync/sync-header-test.js:41]: Again, this will not run on our OSS PRs, even though it's a component test so it could. The setup in `beforeEach` has everything you should need to run this in OSS instead so this module could be renamed to `ent`

**Review by hashishaw — COMMENTED** [review]
Great notes and improvements! A couple ideas & suggestions 👇

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (noelledaley) [reply ui/tests/integration/components/sync/secrets/page/overview-test.js:68]: > Otherwise the test suite just says "ok" unless it fails
oh, that's a great point. i'm used to `cmd + F`ing in test files to figure out the failure and what was happening leading up to it. let's talk about this during our team pattern chat today!

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (noelledaley) [reply ui/tests/integration/components/sync/secrets/page/overview-test.js:86]: ah, that's super helpful to know. i wondered if that might be the case, thanks for confirming!

i could use more clarity on our testing strategy for OSS vs. enterprise. i keep having to remind myself that some tests are expected to fail if i'm running tests locally with an enterprise binary. let's talk about this during the patterns chat too; i haven't quite figured out exactly what question i'd like to ask here but i suspect a more general chat will help. 😅

Comment (hellobontempo) [reply ui/tests/integration/components/sync/secrets/page/overview-test.js:86]: I'm curious which tests were failing locally with the enterprise binary! If you could mention in our discussion later.

File: ui/tests/acceptance/sync/secrets/overview-test.js
Comment (noelledaley) [reply ui/tests/acceptance/sync/secrets/overview-test.js]: agreed! just added coverage for clicking on create & verifying the URL 😀

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (noelledaley) [reply ui/tests/integration/components/sync/secrets/page/overview-test.js:90]: this value is inherited from the top level, but i'll add it again here for clarity

File: ui/tests/integration/components/sync/secrets/page/overview-test.js
Comment (noelledaley) [reply ui/tests/integration/components/sync/secrets/page/overview-test.js]: @hellobontempo you know, i could've sworn i saw a few integration tests fail locally but now that i have my branch in good shape, the only failure i'm repeatedly seeing locally is this acceptance test:
Screenshot 2024-03-28 at 9 24 00 PM

i suppose the failing integration tests i saw earlier was something wrong with my branch, but this test in particular is completely unrelated. 😮

File: ui/tests/acceptance/sync/secrets/overview-test.js
Comment (noelledaley) [inline ui/tests/acceptance/sync/secrets/overview-test.js:78]: this was a funny hack to implement. my gut says we should fix up the `/activation-flags/` mirage handlers to make actual changes to the schema so that the `get` response is updated after the `post` response is called, but this is good enough for now 😬

File: ui/tests/acceptance/sync/secrets/overview-test.js
Comment (noelledaley) [inline ui/tests/acceptance/sync/secrets/overview-test.js:81]: i don't know if an empty object is the right thing to return when no destinations are found, just guessing here 😬

Comment (hashishaw) [reply ui/tests/acceptance/sync/secrets/overview-test.js:81]: It doesn't look like this endpoint is actually called unless we're calling destinations by type. From looking in the UI, when no destinations exist the response to `http://localhost:8200/v1/sys/sync/destinations?list=true` is 
```
{ 
  data: {
    total_destinations: 0
  }
}
```

**Review by hashishaw — APPROVED** [review]
Some minor comments, otherwise LGTM

**Pull request merged**
